Dockerfile

```diff
@@ -64,7 +63,7 @@ else
   )
 fi
 find ${TARGET} -type f -name '*.html' | while read i; do
-  sed -i 's#\(<a[^>]* href="\)https://${DOMAIN}/#\1/#g' "$i"
+  sed -i 's#\(<a[^>]* href="\)${DOCS_URL}/#\1/#g' "$i"
```

Comment:
Let me check this one; I think the intent here was to remove some hard-coded URLs in the docs (e.g. from upstream MarkDown files) that have a full `https://docs.docker.com/` URL in them

URLs in the pages _generated_ by Jekyll are expected to all be relative (so `<a href=/foo/bar/>`, without a domain); the intent of that is so that the generated pages can be deployed to "any" domain (staging/local/elsewhere) without links going to the "live" domain.

So perhaps that part was intentionally set to `docs.docker.com`

As to the `sitemap.xml` having `docs.docker.com` hard-coded; that may also have been intentional at the time, as we've had cases where Google found preview sites, and (although we set `noindex` headers), we wanted to avoid it finding the `sitemap.xml`, and as a result indexing those preview domains (and as a "best effort", we just pointed it to the live website, which was the "least bad" option.